src/pages/workspaces/workspace/RequestAccessModal.js

```diff
@@ -0,0 +1,103 @@
+import _ from 'lodash/fp'
+import { useState } from 'react'
+import { div, h, table, tbody, td, th, thead, tr } from 'react-hyperscript-helpers'
+import { buttonPrimary } from 'src/components/common'
+import { centeredSpinner } from 'src/components/icons'
+import Modal from 'src/components/Modal'
+import { Ajax, useCancellation } from 'src/libs/ajax'
+import { withErrorReporting } from 'src/libs/error'
+import { cond, useOnMount, withBusyState } from 'src/libs/utils'
+
+
+export const RequestAccessModal = ({ onDismiss, workspace }) => {
+  const [groups, setGroups] = useState([])
+  const [accessInstructions, setAccessInstructions] = useState([])
+  const [loadingGroups, setLoadingGroups] = useState(false)
+  const [loadingAccessInstructions, setLoadingAccessInstructions] = useState(false)
+  const signal = useCancellation()
+
+  const { Groups, Workspaces } = Ajax(signal)
+
+  const fetchGroups = _.flow(
+    withBusyState(setLoadingGroups),
+    withErrorReporting('Error loading groups')
+  )(async () => {
+    setGroups(await Groups.list())
+  })
+
+  const fetchAccessInstructions = _.flow(
+    withBusyState(setLoadingAccessInstructions),
+    withErrorReporting('Error loading groups')
+  )(async () => {
+    const i = await Workspaces.workspace(workspace.workspace.namespace, workspace.workspace.name).accessInstructions()
+    console.log(i)
+    setAccessInstructions(i)
+  })
+
+  useOnMount(() => {
+    fetchGroups()
+    fetchAccessInstructions()
+  })
+
+  const loading = loadingGroups || loadingAccessInstructions
+  const groupNames = _.map('groupName', groups)
+  const authDomain = workspace.workspace.authorizationDomain
+  return h(Modal, {
+    title: 'Request Access',
+    width: '40rem',
+    showCancel: false,
+    onDismiss
+  }, [
+    div(['You cannot access this Workspace because it contains restricted data. You need permission from the admin(s) of all of the Groups in the Authorization Domain protecting the workspace.']),
+    loading ?
+      centeredSpinner({ size: 32 }) :
+      table({ style: { margin: '1rem', width: '100%' } }, [
+        thead([
+          tr({ style: { height: '2rem' } }, [th({ style: { textAlign: 'left' } }, ['Group Name']), th({ style: { textAlign: 'left', width: '15rem' } }, ['Access'])])
+        ]),
+        tbody(
+          _.map(({ membersGroupName: groupName }) => tr({ style: { height: '2rem' } }, [
+            td([groupName]),
+            td([
+              _.includes(groupName, groupNames) ?
+                'Yes' :
```

Comment:
TOL: add a little styling to the confirmation `Yes` and possibly rephrase to "Access granted".

Reply:
I don't love "Yes" or "Access granted". The column is labeled "Access". Maybe just "Granted" or "Allowed"? I'd prefer some word that bridges the concepts of access/permission and group membership but I can't think of one; I just keep coming back to a general affirmative statement like "Yes".

Also, what kind of styling did you have in mind?

Reply:
For styling I was thinking of making the "Yes" bold (or italic, etc.) to give it a visual distinction since currently it has the same style as the name of the Auth domain group.
For the wording, I'm also on the fence about it and I'm fine with the simple "Yes".

Reply:
Went with bold.